xdesign/geometry.py

```diff
@@ -205,6 +210,9 @@ def midpoint(self, other):
         """Returns the midpoint between entities."""
         return self.distance(other) / 2.
 
+    def generate(self, grid, material):
+        pass
```

Comment:
It's fine if you want to add a placeholder, but it needs a docstring to tell what it is supposed to be placeholding.